docs/userguide/networking/dockernetworks.md

```diff
@@ -0,0 +1,368 @@
+
+
+### An overlay network
+
+Docker's `overlay` network driver supports multi-host networking natively
+out-of-the-box. This support is accomplished with the help of `libnetwork`, a
+built-in VXLAN-based overlay network driver, and Docker's `libkv` library.
+
+The `overlay` network requires a valid key-value store service. Currently, Docker's supports Consul, Etcd, Zookeeper (Distributed store) and
+BoltDB (Local store). Before creating a network you must install and configure your chosen key-value store service.  The Docker hosts that you intend to network and the service must be able to communicate.
+
+![](images/key_value.png)
+
+Each host in the network must run a Docker Engine instance. The easiest way to provision the hosts are with Docker Machine.
+
+![](images/engine_on_net.png)
+
+Using Docker Swarm you can quickly create a cluster which includes a discovery service as well.
+
+To create an overlay network, you configure options on  the `daemon` on each Docker Engine for use with `overlay` network. There are two options to set:
+
+| Option                           | Description                                               |
+|----------------------------------|-----------------------------------------------------------|
+| `--cluster-store=PROVIDER://URL` | Describes the location of the KV service.                 |
+| `--cluster-advertise=HOST_IP`    | Advertises containers created by the HOST on the network. |
+
+Create an `overlay` network on one of the machines in the Swarm.
+
+        $ docker network create -d overlay my-multi-host-network
+
+This results in a single network spanning multiple hosts. The `overlay` network provides complete isolation for the containers.
+
+![](images/overlay_network.png)
+
+Then, on each host, launch containers making sure to specify the network name.
+
+        $ docker run run -itd --net=mmy-multi-host-network busybox
+
+Once launched, each container has access to all the containers in the network regardless of which Docker host the container was launched on.
+
+![](images/overlay-network-final.png)
+
+If you would like to try this for yourself, see the [Getting started for overlay](get-started-overlay.md).
+
+### Custom network plugin
+
+If you like, you can create your own custom, network driver plugin. A network
+driver plugin makes use of Docker's plugin infrastructure. In this
+infrastructure, a plugin is a process running on the same Docker host as the
+Docker `daemon`.
+
+Network plugins follow the same restrictions and installation rules as other
+plugins. All plugins make use of the plugin API. They have a lifecycle that encompasses installation, starting, stopping and activation.
+
+Once you have creatd a custom network driver, you use it like the built-in
+network drivers. For example,
+
+    $ docker network create -d weave mynet
```

Comment:
(Mentioned earlier) I'd prefer using the long form `--driver`, as it's more descriptive